### PR TITLE
Revert "Revert "fix(tracing): fix msgpack import error""

### PR DESCRIPTION
**Pull request by mabdinur**
Reverts: https://github.com/DataDog/dd-trace-py/pull/7277 and lints files to pass ci.

Note - there is a chain of reversions here. https://github.com/DataDog/dd-trace-py/pull/7277 (ci fix) reverted https://github.com/DataDog/dd-trace-py/commit/fe6c74ff4e6de5b726393ad618318cc64ba6cd3a (encoding fix) which reverted https://github.com/DataDog/dd-trace-py/pull/7099 (feature to debug v05 encoding). Here we are rolling back to the first revert.

# PR Description for OG fix

There is an edge case were the string table used by the v0.5 encoder swaps entries. This corrupts trace payloads. In this PR I will add debug logs that will help us reproduce the behavior that triggered incident-24455.

RISK
This PR maybe increase memory usage when v0.5 is enabled. With this change we will hold a reference to a all encoded spans until the encoded bytes are flushed and we will log trace payloads that can have a size of up to 50MB when we traces are misencoded.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: tests/tracer/test_encoders.py [hunk 1]
```diff
@@ -795,139 +792,3 @@ def test_json_encoder_traces_bytes():
         assert "\ufffdspan.a" == span_a["name"], span_a["name"]
         assert "\x80span.b" == span_b["name"]
         assert "\ufffdspan.b" == span_c["name"]
-
-
-@pytest.mark.skipif(
-    os.getenv("PYTHONOPTIMIZE", "").lower() in ("1", "t", "true"),
-    reason="Python optimize removes assertions from cython code",
-)
-def test_verifying_v05_payloads():
```
Comment (majorgreys) [inline tests/tracer/test_encoders.py:804]: Why is this test being dropped?

File: tests/integration/test_integration.py [hunk 2]
```diff
@@ -533,31 +533,6 @@ def encode_traces(self, traces):
...
 
-@skip_if_testagent
-@pytest.mark.subprocess(
-    env={
-        "_DD_TRACE_WRITER_LOG_ERROR_PAYLOADS": "true",
-        "DD_TRACE_API_VERSION": "v0.5",
-        "DD_TRACE_WRITER_INTERVAL_SECONDS": "1000",
-    }
-)
-def test_trace_with_invalid_encoding_for_v05_payload():
```
Comment (majorgreys) [inline tests/integration/test_integration.py:544]: Why delete this?

File: ddtrace/internal/_encoding.pyx [hunk 3]
```diff
@@ -922,75 +901,6 @@ cdef class MsgpackEncoderV05(MsgpackEncoderBase):
 
         return 0
 
-    cpdef _verify_encoding(self, encoded_bytes):
```
Comment (majorgreys) [inline ddtrace/internal/_encoding.pyx:925]: The only change I see here is that the encoded payload is not verified (which was where msgpack was being imported).

**Review by majorgreys — COMMENTED** [review]
I'm still struggling to understand what the original problem was and what the fix is. I see a lot of code being deleted in this change, so I am not sure what is being added back in?

File: tests/tracer/test_encoders.py (hunk 1 above)
Comment (mabdinur) [reply tests/tracer/test_encoders.py:804]: All this PR does is revert the code that was introduced in this PR and resolves some linting conflicts. There are no additional changes: https://github.com/DataDog/dd-trace-py/pull/7277.

File: tests/integration/test_integration.py (hunk 2 above)
Comment (mabdinur) [reply tests/integration/test_integration.py:544]: Because it tests a feature that is being reverted

File: ddtrace/internal/_encoding.pyx (hunk 3 above)
Comment (mabdinur) [reply ddtrace/internal/_encoding.pyx:925]: This changes reverts verifying encoding. This feature is reliant on the msgpack dependency. By removing this feature we no longer need to import msgpack the import errors disappear.

**Review by majorgreys — APPROVED** [review]

**Comment by mabdinur** [comment]
> I'm still struggling to understand what the original problem was and what the fix is. I see a lot of code being deleted in this change, so I am not sure what is being added back in?

Original problem: ddtrace does NOT install msgpack by default. If someone tries to use v0.5 without installing msgpack manually this line will throw an error https://github.com/DataDog/dd-trace-py/blob/2.x/ddtrace/internal/_encoding.pyx#L926. This bug was introduced a feature that had a lot of issues. Instead of just fixing the msgpack import error this PR reverts the feature. This problematic feature is internal and used to debug trace payloads. It was already released in ddtrace v2.1.0.

**Pull request merged**